### PR TITLE
Adding dbscan code from class on Monday 2/24

### PR DESCRIPTION
**Pull request by nshalash**

### CODE REVIEW

**Comment by gallettilance** [comment]
@nshalash The file is here. It must be in the master branch of your github repo. If you either checkout your master branch of fetch origin/master you should recover this file. Then you should be able to squash your commits and I can review this PR.

File: 04-dbscan/dbscan.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import sim
```
Comment (gallettilance) [inline 04-dbscan/dbscan.py]: this should be
```suggestion
from .sim import euclidean_dist
```

File: 04-dbscan/dbscan.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+class DBC():
+    def __init__(self, dataset, min_pts, epsilon):
+        self.dataset = dataset
+        self.min_pts = min_pts
+        self.epsilon = epsilon
+
+    def _get_neighborhood(self, P):
+        Neighborhood = []
+        for Pn in range(len(self.dataset)):
+            if sim.euclidean_dist(self.dataset[P], self.dataset[Pn]) <= self.epsilon:
```
Comment (gallettilance) [inline 04-dbscan/dbscan.py]:
```suggestion
            if euclidean_dist(self.dataset[P], self.dataset[Pn]) <= self.epsilon:
```

File: 04-dbscan/dbscan.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        # need to find all core points
+    
+        # all core points in same epsilon neighborhood should be assigned to the same cluster
+        assignment = 0
+        for P in range(len(self.dataset)):
+            # if assignment is not 0 here, point has been assigned so we should move on
+            if assignments [P] != 0:
+                continue
+            PNeighborhood = self._get_neighborhood(P)
+            # implement high level functions first, then helper functions
```
Comment (gallettilance) [inline 04-dbscan/dbscan.py]: this comment is probably not needed

Comment (gallettilance) [reply 04-dbscan/dbscan.py]: But is a good practice for you during in class demos!

**Comment by gallettilance** [comment]
@nshalash just following up - please change the location of this file when you get a chance

**Comment by nshalash** [comment]
@gallettilance I moved the file and typed git commit --amend. Now do I need to force push it?

**Comment by nshalash** [comment]
I accidentally added a million commits again...sorry this keeps happening. I really don't know why I keep getting "noop" every time I try to squash commits.

**Pull request closed without merging**